mpeg2/src/pes.rs

```diff
@@ -13,6 +15,7 @@ pub struct PacketizationConfig {
     pub packet_id: u16,
     pub random_access_indicator: bool,
     pub continuity_counter: u8,
+    pub temi: Option<Vec<TEMITimelineDescriptor>>,
```

Comment:
Is there any difference in behavior between `None` and `Some(vec![])`? Why not just make this a `Vec`?

```suggestion
    pub temi: Vec<TEMITimelineDescriptor>,
```

Reply:
By using `Option`, we can simply call `take()` to move the temi vectors into AdaptationField, all the following would be `None`.  
```rust
fn next(&mut self) -> Option<Self::Item> {
    let adaptation_field = self.header.map(|header| {
        let mut af = ts::AdaptationField {
            random_access_indicator: if self.config.random_access_indicator { Some(true) } else { None },
            temi_timeline_descriptors: self.config.temi.take(),
```

Reply:
Let's try to avoid a leaky abstraction here. The implementation details of that function shouldn't dictate what the public API looks like.

There are a few ways you can do the same thing without making this an `Option`. For example, you could `mem::swap` it with an empty `Vec`.

Reply:
Oh, okay, let me check the `mem::swap` method.